app.rb

```diff
@@ -0,0 +1,148 @@
+require_relative './book'
+require_relative './teacher'
+require_relative './student'
+require_relative './person'
+require_relative './rental'
+# App object class
+class App
+  def initialize
+    @book_list = []
+    @rentals = []
+    @person = []
+  end
+
+  def menu
+    puts "Welcome to school library ! \n\n"
+    puts 'Please choose an option by entering a number:'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person id'
+    puts "7 - Exit\n\n"
+    gets.chomp
+  end
+
+  def menu_run
+    choice = menu.to_s
+    case choice
+    when '1'
+      make_books
+    when '2'
+      make_persons
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_all_rentals_for_a_given_person_id
+    else
+      puts 'Thank you for using this App'
+      exit
+    end
+  end
+
+  def create_book
+    print 'Title: '
+    input1 = gets.chomp
+    print 'Author: '
+    input2 = gets.chomp
+    book = Book.new(input1, input2)
+    @book_list.push(book)
+    puts 'Book created successfully '
+    menu_run
+  end
+
+  def create_person
+    print "\n Do you want to create a student (1) or a teacher (2)? [Input the number]: "
+    user_input = gets.chomp
+    case user_input
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
```

Comment:
- [ ] I still cannot add a teacher, I tried to check the reason, and found that you added create_student method but did not add create_teacher method. So when I select create_teacher method, app just cannot find the method to run. Please add create_teacher to make the app work. You can start from checking the code below:  


```
  def create_teacher
    puts 'Enter age: '
    age = gets.chomp
    puts 'Enter name: '
    name = gets.chomp
    puts "'Has parent permission? [Y/N]: '"
    permission = gets.chomp
      teacher = Teacher.new(age, name, parent_permission: false)
      @person.push(teacher)
  end

```

After adding the missing method, please make sure that it works properly (name, age, and etc fields are added in object)